### DISCUSSION TITLE
Button to go to the next record of the data provider

### DISCUSSION THREAD

**Discussion by NicolasFabre81** [Budibase/budibase#9580] [Help] [answered]
In my page, I have a Data provider (sourced on a postgreSQL table) with a one row limit, and a repeater.
I want to display only one record at a time, because there is a lot of fields.
How can I make a link or button to go to the next (or previous) record of the data provider ?

I think I can play with queries and LIMIT / OFFSET as in https://github.com/Budibase/budibase/discussions/7313 but do you see a smarter way ?

**Comment by andz-bb** [comment]
hey @NicolasFabre81

For the simplest approach could you use the built in pagination option on the Data Provider to achieve what you're looking for? If you need something more custom than that then the query approach might be the best way to go 👍

**Comment by abadedgar113** [comment, marked as answer]
Hello @NicolasFabre81 

I use queries with LIMIT and OFFSET then combine it with query params.

Here is my sample query:
<img width="1016" alt="Screenshot 2025-07-09 at 10 01 15 AM" src="https://github.com/user-attachments/assets/e96a1421-f0c4-4c30-831b-a0d163a92ea8" />
And this is my javascript code in offset binding:
<img width="1022" alt="Screenshot 2025-07-09 at 10 02 33 AM" src="https://github.com/user-attachments/assets/f5b09f42-09f8-43ac-83c4-d25cbfd3542f" />

For the next button, i use Navigate To action then add this simple javascript 
`let page = 1;

page = page + 1;

return "/screenName?page=" + page;`
